shale/src/lib.rs

```diff
@@ -93,43 +93,18 @@ pub trait CachedStore: Debug + Send + Sync {
     fn id(&self) -> SpaceId;
 }
 
-/// A addressed, typed, and read-writable handle for the stored item in [ShaleStore]. The object
-/// represents the decoded/mapped data. The implementation of [ShaleStore] could use [ObjCache] to
-/// turn a `TypedView` into an [ObjRef].
-pub trait TypedView<T: ?Sized + Send + Sync>:
-    std::fmt::Debug + Deref<Target = T> + Send + Sync
-{
-    /// Get the offset of the initial byte in the linear space.
-    fn get_offset(&self) -> usize;
-    /// Access it as a [CachedStore] object.
-    fn get_mem_store(&self) -> &dyn CachedStore;
-    /// Access it as a mutable CachedStore object
-    fn get_mut_mem_store(&mut self) -> &mut dyn CachedStore;
-    /// Estimate the serialized length of the current type content. It should not be smaller than
-    /// the actually length.
-    fn estimate_mem_image(&self) -> Option<u64>;
-    /// Serialize the type content to the memory image. It defines how the current in-memory object
-    /// of `T` should be represented in the linear storage space.
-    fn write_mem_image(&self, mem_image: &mut [u8]) -> Result<(), ShaleError>;
-    /// Gain mutable access to the typed content. By changing it, its serialized bytes (and length)
-    /// could change.
-    fn write(&mut self) -> &mut T;
-    /// Returns if the typed content is memory-mapped (i.e., all changes through `write` are auto
-    /// reflected in the underlying [CachedStore]).
-    fn is_mem_mapped(&self) -> bool;
-}
 
 /// A wrapper of `TypedView` to enable writes. The direct construction (by [Obj::from_typed_view]
 /// or [StoredView::ptr_to_obj]) could be useful for some unsafe access to a low-level item (e.g.
 /// headers/metadata at bootstrap or part of [ShaleStore] implementation) stored at a given [DiskAddress]
 /// . Users of [ShaleStore] implementation, however, will only use [ObjRef] for safeguarded access.
 #[derive(Debug)]
-pub struct Obj<T: ?Sized + Send + Sync> {
-    value: Box<dyn TypedView<T>>,
+pub struct Obj<T: Storable> {
+    value: Box<StoredView<T>>,
```

Comment:
I think you can remove the `Box` now. Up to you whether you want to do it in this PR or a follow up.

Reply:
Follow up! But yeah it can probably be removed.